### DISCUSSION TITLE
EJBCA Finalize Installation Adminweb not available at https://localhost:8443/ejbca/adminweb/

### DISCUSSION THREAD

**Discussion by JannikZorn** [Keyfactor/ejbca-ce#270] [Q&A] [answered]
Hello,

my current setup is:
ubuntu 20.04
wildfly 22.0.0
openjdk 11.0.18
javadoc 11.0.18

I followed the instructions of the official website for the installation (https://doc.primekey.com/ejbca/ejbca-installation) but got stucked at the "Finalize Installation" part.
I passed the "Deploying" part with SUCCESS and without errors in the server.log.
![Unbenannt4](https://user-images.githubusercontent.com/119753966/232055919-1f007589-53c3-4f12-bc59-1f80d6bcf271.PNG)

I also passed the "Installation" part with SUCCESS output. 
![Unbenannt3](https://user-images.githubusercontent.com/119753966/232056050-81dc70a8-cb28-4418-bde3-990cd2a90a35.PNG)

wildfly service is up and running
![Unbenannt2](https://user-images.githubusercontent.com/119753966/232056135-d61f3df4-9866-429e-9aeb-5a53c1be7533.PNG)

But now I'm stuck at the "Finalize" part. I can't open the admin UI at https://localhost:8443/ejbca/adminweb/.
I tried with the IP Adress of the server and also checked the open ports but port 8443 seems not to be open.
I checked the server.log again and saw the admin web is listening to a different port. But still same. I can't access the admin UI.
I get the Error ERR_CONNECTION_REFUSED.

![Unbenannt](https://user-images.githubusercontent.com/119753966/232057270-44796855-dabb-463c-83f6-6903ecfb1255.PNG)

Do you have any idea how to fix it?
Thanks & best regards

**Comment by primetomas** [comment]
Did you import superadmin.p12 into your web browser?

**Reply by primetomas** [reply]
to check that the application is up, you can go to https://localhost:8442/ejbca (only TLS server authentication)

Also see wildfly standalone/log/server.log for possible errors during startup.

**Reply by JannikZorn** [reply]
Not yet. I don't know how to.
I used the docker image before. There I was able to open a website and download and import the superadmin.p12 from the website.

**Reply by primetomas** [reply]
After running "ant install" superadmin.p12 is generated in the p12 subdirectory.

**Reply by JannikZorn** [reply]
I imported the superadmin.p12 now and also tried to access  https://localhost:8442/ejbca but still ERR_CONNECTION_REFUSED.
There were no errors during the deploy/installation process in server.log.
Also in the server.log it shows Admin console listening on http://127.0.0.1:9990/ but also this port doesn't work

**Comment by JannikZorn** [comment]


**Reply by primetomas** [reply]
restart wildfly and check the log during startup.

**Reply by primetomas** [reply]
assuming you did all the TLS setup configuration in https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-HTTP(S)Configuration?

Looks like you are using WIldFly 22? Is that correct? 
The instruction here currently have for 24, 25 and 26. https://doc.primekey.com/ejbca/ejbca-installation/application-servers

**Reply by JannikZorn** [reply]
assuming you did all the TLS setup configuration in https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-HTTP(S)Configuration?
Yes I did

Looks like you are using WIldFly 22? Is that correct?
Yes Wildfly 22.0.0

Restart wildfly and check the log during startup.
I did and I found 2 errors in the server log:

2023-04-17 11:10:27,144 ERROR [org.jboss.as.jsf] (MSC service thread 1-1) WFLYJSF0002: Could not load JSF managed bean class: org.ejbca.ui.web.admin.peerconnector.PeerConnectorsMBean
2023-04-17 11:10:27,145 ERROR [org.jboss.as.jsf] (MSC service thread 1-1) WFLYJSF0002: Could not load JSF managed bean class: org.ejbca.ui.web.admin.peerconnector.PeerConnectorMBean


2023-04-17 11:10:30,489 INFO  [org.apache.commons.beanutils.FluentPropertyBeanIntrospector] (MSC service thread 1-5) Error when creating PropertyDescriptor for public final void org.apache.com>

**Reply by JannikZorn** [reply]
Should I try to use a different wildfly version?

**Reply by primetomas** [reply]
If your server logs shows some error, or does not show that TLS on port 8442 and 8443 have been initialized. Yes I suggest that. I suggest using WildFly 26 together with EJBCA 7.11.

**Reply by primetomas** [reply]
This errors you past above are not actually errors, and are completely fine btw.

**Reply by JannikZorn** [reply]
I did a reinstalltion now with wildfly 26. now the error is different ERR_SSL_PROTOCOL_ERROR

**Reply by JannikZorn** [reply]
I imported the superadmin.p12 already

**Reply by JannikZorn** [reply]
On http://localhost:8080/ejbca/ I see now the EJBCA page, So I guess there is something wrong with the superadmin.p12 cert

**Reply by JannikZorn** [reply]
I tried a completly new installation but I'm still unable to acess adminweb. COuld anyone help me out? I tried the IP adress with port 8443 as well

**Comment by srasay2** [comment]
What browser are you using and are you connecting in an incognito session?

Firefox is the recommended browser for ease of importing your keys and
setting them up to gain access to the Admin Web application.

R

On Thu, Apr 27, 2023 at 01:26 JannikZorn ***@***.***> wrote:

> I tried a completly new installation but I'm still unable to acess
> adminweb. COuld anyone help me out? I tried the IP adress with port 8443 as
> well
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Keyfactor/ejbca-ce/discussions/270#discussioncomment-5740041>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABFMQGNCNGQ77QSFAUDIH3XDIURXANCNFSM6AAAAAAW6NYZEE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Rich Asay

**Reply by JannikZorn** [reply]
What browser are you using and are you connecting in an incognito session?
I tried with Firefox 112.0.2 (64-Bit) and  in an incognito session.
I imported the superadmin.p12
but still SSL Handshake ERROR

I found the following discussion and tried to follow the troubleshooting steps:
https://github.com/Keyfactor/ejbca-ce/discussions/46

but converting the keystore und truststore to .p12 and restarting wildfly only caused ERR_CONNECTION_REFUSED

**Reply by JannikZorn** [reply, marked as answer]
I solved it now it was the same issue as discussed in https://github.com/Keyfactor/ejbca-ce/discussions/46
The standalone.xml still tries to use kestore.p12 and truststore.p12 but can't find those files because the files called keystore.jks and truststore.jks. I used wildfly-26.0.0 and the documentation wasn't updated yet.
 Solution was the following:

cd /opt/wildfly/standalone/configuration/keystore
keytool -importkeystore -srckeystore keystore.jks -srcstoretype JKS -deststoretype PKCS12 -destkeystore keystore.p12
keytool -importkeystore -srckeystore truststore.jks -srcstoretype JKS -deststoretype PKCS12 -destkeystore truststore.p12

systemctl restart wildfly

Thanks for your help anyway!
